### PR TITLE
Handle types that are not field metadata

### PR DESCRIPTION
**Pull request by thomtrp**
Some steps that do not return field metadata types need to be handled

### CODE REVIEW

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 1]
```diff
@@ -534,17 +534,486 @@ describe('evaluateFilterConditions', () => {
...
+            null,
+            '',
+            'CUSTOM_ARRAY',
+          );
+          const filter2 = createFilter(
+            ViewFilterOperand.IsEmpty,
+            [],
+            '',
+            'CUSTOM_ARRAY',
+          );
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts]: TBH I don't get why that is expected to return false

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 2]
```diff
@@ -534,17 +534,486 @@ describe('evaluateFilterConditions', () => {
...
+          const filter3 = createFilter(
+            ViewFilterOperand.Is,
+            obj1,
+            obj3,
+            'CUSTOM_OBJECT',
+          );
+
+          expect(evaluateFilterConditions({ filters: [filter1] })).toBe(true); // same reference
+          expect(evaluateFilterConditions({ filters: [filter2] })).toBe(false); // different reference
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts]: same, this should return true isn't it?

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 3]
```diff
@@ -534,17 +534,486 @@ describe('evaluateFilterConditions', () => {
...
+            null,
+            '',
+            'CUSTOM_OBJECT',
+          );
+          const filter2 = createFilter(
+            ViewFilterOperand.IsEmpty,
+            { name: 'John' },
+            '',
+            'CUSTOM_OBJECT',
+          );
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts]: please add a test with 
```suggestion
          const filter2 = createFilter(
            ViewFilterOperand.IsEmpty,
            { name: 'John' },
            '',
            'CUSTOM_OBJECT',
          );
          
           const filter3 = createFilter(
            ViewFilterOperand.IsEmpty,
            {},
            '',
            'CUSTOM_OBJECT',
          );
```

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 4]
```diff
@@ -534,17 +534,486 @@ describe('evaluateFilterConditions', () => {
...
+            ViewFilterOperand.Contains,
+            obj,
+            'John',
+            'CUSTOM_OBJECT',
+          );
+
+          // Objects are converted to '[object Object]' string
+          expect(evaluateFilterConditions({ filters: [filter1] })).toBe(true);
+          expect(evaluateFilterConditions({ filters: [filter2] })).toBe(false);
+        });
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts]: ? o_0

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 5]
```diff
@@ -534,17 +534,486 @@ describe('evaluateFilterConditions', () => {
...
+            null,
+            undefined,
+            'MIXED_TYPE',
+          );
+          const filter2 = createFilter(
+            ViewFilterOperand.Is,
+            null,
+            null,
+            'MIXED_TYPE',
+          );
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts]: add a filter3 test to check it returns false if something not null is given to filter

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 6]
```diff
@@ -534,17 +534,486 @@ describe('evaluateFilterConditions', () => {
...
+            ViewFilterOperand.Is,
+            false,
+            '',
+            'MIXED_TYPE',
+          );
+
+          expect(evaluateFilterConditions({ filters: [filter1] })).toBe(false); // true != 'true' (loose equality)
+          expect(evaluateFilterConditions({ filters: [filter2] })).toBe(true); // false == ''
+        });
+      });
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts]: really? don't really get that either

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 7]
```diff
@@ -534,17 +534,173 @@ describe('evaluateFilterConditions', () => {
...
+          'unknown',
+        );
+        const filter2 = createFilter(
+          ViewFilterOperand.Is,
+          'test',
+          'different',
+          'unknown',
+        );
+
+        expect(evaluateFilterConditions({ filters: [filter1] })).toBe(true);
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:577]: I think we should add edge cases undefined and null for all tests. It will probably avoid tough debug investigations

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 8]
```diff
@@ -534,17 +534,173 @@ describe('evaluateFilterConditions', () => {
...
+      });
+
+      it('should handle IsEmpty operand with unknown type', () => {
+        const filter1 = createFilter(
+          ViewFilterOperand.IsEmpty,
+          null,
+          '',
+          'unknown',
+        );
+        const filter2 = createFilter(
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:772]: Add a test case with an empty string ""

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 9]
```diff
@@ -534,17 +534,173 @@ describe('evaluateFilterConditions', () => {
...
+      });
+
+      it('should handle IsNotEmpty operand with unknown type', () => {
+        const filter1 = createFilter(
+          ViewFilterOperand.IsNotEmpty,
+          'not empty',
+          '',
+          'unknown',
+        );
+        const filter2 = createFilter(
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:811]: Same

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 10]
```diff
@@ -534,17 +534,173 @@ describe('evaluateFilterConditions', () => {
...
+          50,
+          'unknown',
+        );
+        const filter2 = createFilter(
+          ViewFilterOperand.GreaterThanOrEqual,
+          30,
+          50,
+          'unknown',
+        );
+
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:869]: Add test case With string numbers

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 11]
```diff
@@ -534,17 +534,173 @@ describe('evaluateFilterConditions', () => {
...
+      });
+
+      it('should handle LessThanOrEqual operand with unknown type', () => {
+        const filter1 = createFilter(
+          ViewFilterOperand.LessThanOrEqual,
+          30,
+          50,
+          'unknown',
+        );
+        const filter2 = createFilter(
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:883]: Same

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 12]
```diff
@@ -534,17 +534,173 @@ describe('evaluateFilterConditions', () => {
...
+    describe('unknown type filters', () => {
+      it('should handle Is operand with unknown type', () => {
+        const filter1 = createFilter(
+          ViewFilterOperand.Is,
+          'test',
+          'test',
+          'unknown',
+        );
+        const filter2 = createFilter(
+          ViewFilterOperand.Is,
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:559]: Does all operand are tested? I think they should be

Comment (thomtrp) [reply packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:559]: yes

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 13]
```diff
@@ -534,17 +534,328 @@ describe('evaluateFilterConditions', () => {
...
+          'unknown',
+        );
+        const filter4 = createFilter(
+          ViewFilterOperand.Contains,
+          [1, 2, 3],
+          4,
+          'unknown',
+        );
+        const filter5 = createFilter(
+          ViewFilterOperand.DoesNotContain,
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts]:
```suggestion
          ViewFilterOperand.Contain,
```

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 14]
```diff
@@ -534,17 +534,328 @@ describe('evaluateFilterConditions', () => {
...
+          null,
+          null,
+          'unknown',
+        );
+        const filter6 = createFilter(
+          ViewFilterOperand.DoesNotContain,
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts]:
```suggestion
          ViewFilterOperand.Contain,
```

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 15]
```diff
@@ -534,17 +534,328 @@ describe('evaluateFilterConditions', () => {
...
+        expect(evaluateFilterConditions({ filters: [filter4] })).toBe(false);
+      });
+
+      it('should handle Contains operand with unknown type', () => {
+        const filter1 = createFilter(
+          ViewFilterOperand.Contains,
+          'Hello World',
+          'World',
+          'unknown',
+        );
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:621]: add test when 

```
createFilter(
          ViewFilterOperand.Contains,
          'Hello World',
          undefined,
          'unknown',
        );

createFilter(
          ViewFilterOperand.Contains,
          'Hello World',
          null,
          'unknown',
        );

```

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 16]
```diff
@@ -534,17 +534,328 @@ describe('evaluateFilterConditions', () => {
...
+        const filter2 = createFilter(
+          ViewFilterOperand.Contains,
+          'Hello',
+          'World',
+          'unknown',
+        );
+        const filter3 = createFilter(
+          ViewFilterOperand.Contains,
+          [1, 2, 3],
+          2,
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:631]: same, 2 other cases with undefined and null

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 17]
```diff
@@ -534,17 +534,328 @@ describe('evaluateFilterConditions', () => {
...
+          'unknown',
+        );
+        const filter6 = createFilter(
+          ViewFilterOperand.DoesNotContain,
+          undefined,
+          undefined,
+          'unknown',
+        );
+
+        expect(evaluateFilterConditions({ filters: [filter1] })).toBe(true);
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:678]: for better readability, I would update `createFilter` to `createFilterTestCase`
```
const testCase1 = createFilterTestCase({filterParams: ..., expected:Result} -> {filterCondition: ...; expectedResult}
```

and then check 
```
expect(evaluateFilterConditions(testCase1.filterCondition).toBe(testCase1.expectedResult);
```

It would be far either to read and maintain to be honest

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 18]
```diff
@@ -534,17 +534,328 @@ describe('evaluateFilterConditions', () => {
...
+        const filter1 = createFilter(
+          ViewFilterOperand.DoesNotContain,
+          'Hello',
+          'World',
+          'unknown',
+        );
+        const filter2 = createFilter(
+          ViewFilterOperand.DoesNotContain,
+          'Hello World',
+          'World',
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:700]: same, add undefined and null

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts [hunk 19]
```diff
@@ -534,17 +534,328 @@ describe('evaluateFilterConditions', () => {
...
+        const filter3 = createFilter(
+          ViewFilterOperand.DoesNotContain,
+          [1, 2, 3],
+          2,
+          'unknown',
+        );
+        const filter4 = createFilter(
+          ViewFilterOperand.DoesNotContain,
+          [1, 2, 3],
+          4,
```
Comment (martmull) [inline packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:712]: same

File: packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts (hunk 17 above)
Comment (thomtrp) [reply packages/twenty-server/src/modules/workflow/workflow-executor/workflow-actions/filter/utils/__tests__/evaluate-filter-conditions.util.spec.ts:678]: I don't want to refacto everything now. I spent enough time on those tests :)

**Review by martmull — APPROVED** [review]
